ingestion_tools/scripts/data_validation/conftest.py

```diff
@@ -35,35 +36,56 @@ def run_names(bucket: str, dataset: str, run_glob: str) -> List[str]:
     return run_names
 
 
-def voxel_spacings(bucket: str, dataset: str, voxelspacing_glob: str) -> List[float]:
-    """All voxel spacings matching the vozel spacing glob pattern in the dataset."""
+def get_voxel_spacing_files(bucket: str, dataset: str, run_names: List[str], voxelspacing_glob: str) -> List[str]:
     fs: S3Filesystem = FileSystemApi.get_fs_api(mode="s3", force_overwrite=False)
-    tentatives = fs.glob(f"s3://{bucket}/{dataset}/*/Tomograms/{voxelspacing_glob}")
+
+    tentatives = []
+
+    # Use ThreadPoolExecutor to run glob_task in parallel
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        # Map the run_names to the glob_task function
+        results = executor.map(
+            lambda run_name: fs.glob(f"s3://{bucket}/{dataset}/{run_name}/Tomograms/{voxelspacing_glob}"),
```

Comment:
nit: could we create a formatted string outside the lambda, which already has the bucket, dataset, voxelspacing_glob applied? So, that this formatting only has to apply the run name?

Reply:
done in #250!